icu4c/source/i18n/usearch.cpp

```diff
@@ -317,7 +317,7 @@ inline uint16_t initializePatternCETable(UStringSearch *strsrch,
         uprv_free(pattern->ces);
     }
 
-    uint16_t  offset      = 0;
+    uint32_t  offset      = 0;
```

Comment:
Hmm. There is inconsistency in the places this offset value is used; some are int32_t (like pattern->cesLength, pattern->pcesLength) and some are uint32_t (like the corresponding parameter in addTouint64_tArray). We may need to be more consistent about usage.

Reply:
Thanks Peter. I could file a follow up ticket for 66 or 67, and this ticket could be just for the maint-65 release.

Reply:
I filed a follow-up ticket: https://unicode-org.atlassian.net/browse/ICU-20833

Aside: Since `addTouint64_tArray` uses `uint32_t` I wonder if we'd want to use that everywhere in here?

Reply:
Yeah, probably uint32_t everywhere is best. Thanks for the followup! Will approve this one for 65.